### ISSUE TITLE
Calling createDataSourceFromS3Async results in "Operation not found" error

### ISSUE THREAD

**Issue by wjgilmore** [aws/aws-sdk-php#1346]
Hi,

We've been using the AWS SDK for PHP for quite some time now. It has so far worked flawlessly however today I'm running into a strange issue associated with the Machine Learning API. When I run the following code I receive the error `Operation not found: CreateDataSourceFromS3Async`:

    $s3 = AWS::createClient('s3');
    $s3->createDataSourceFromS3Async(
	    [
	        'ComputeStatistics' => true,
	        'DataSourceId' => "<removed>",
	        'DataSourceName' => "<removed>",
	        'DataSpec' => [
	            'DataLocationS3' => "<removed>",
	            'DataRearrangement' => "<removed>",
	            'DataSchema' => "<removed>",
	        ],
	    ]
	);

Incidentally the constructor looks a tad different than that used within the actual SDK  only because I'm using the SDK by way of the Laravel service provider (https://github.com/aws/aws-sdk-php-laravel). This pulls in the 3.X SDK.

I can use all other APIs just fine: `putObjectAsync`, `getObject`, etc. It's just the Machine Learning-specific calls which are identified as not found. Any ideas?

Jason

**Comment by kstich** [comment]
@wjgilmore Can you tell us which version of the AWS SDK has been pulled in for your use here? I'm able to get past the ['Operation not found' check in `AwsClient`](https://github.com/aws/aws-sdk-php/blob/master/src/AwsClient.php#L215) for the `'CreateDataSourceFromS3Async'` operation.

**Comment by wjgilmore** [comment]
Sure thing. My project is using aws-sdk-php version 3.32.4. When I run the specific snippet from original message, I see:

    InvalidArgumentException: Operation not found: CreateDataSourceFromS3 in 
    /home/vagrant/dev.example.com/vendor/aws/aws-sdk-php/src/AwsClient.php:215

Really strange since all other APIs work perfect.

**Comment by wjgilmore** [comment]
Sigh I stand corrected. The aws-sdk-php-laravel package `composer.json` file indicates a `~3.0` release is used, which I suppose is the source of the issue because I'd imagine an AWS SDK PHP 3.0.X release did not yet include the ML API:

    "aws/aws-sdk-php": "~3.0",

I don't see a tagged 3.0.X version being available at https://github.com/aws/aws-sdk-php so I can't confirm this (I'm sure there is some way). But in the meantime I'm just going to call the AWS SDK for PHP directly from my Laravel application and move on with life.

Thank you,
Jason

**Comment by wjgilmore** [comment]
Indeed you are correct regarding Composer and the tilde, sorry for the mixup. However, a moment ago I just realized my problem. The code snippet found in my original report was incorrect:

    $s3 = AWS::createClient('s3');
    $s3->createDataSourceFromS3Async(
    ...
    );

The `createDataSourceFromS3Async` method is of course not available to the S3 client. It is part of the MachineLearning client. :-)

I made this correction and everything is working perfectly. Sorry about the false alarm.

Jason
